### ISSUE TITLE
Improve "'await' has no effect on the type of this expression" hints in JavaScript when extending Promise

### ISSUE THREAD

**Issue by dnicolson** [microsoft/TypeScript#57531]
### 🔍 Search Terms

'await' has no effect on the type of this expression, 80007, extends, promise

### ✅ Viability Checklist

- [X] This wouldn't be a breaking change in existing TypeScript/JavaScript code
- [X] This wouldn't change the runtime behavior of existing JavaScript code
- [X] This could be implemented without emitting different JS based on the types of the expressions
- [X] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, new syntax sugar for JS, etc.)
- [X] This isn't a request to add a new utility type: https://github.com/microsoft/TypeScript/wiki/No-New-Utility-Types
- [X] This feature would agree with the rest of our Design Goals: https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals

### ⭐ Suggestion

Consider the following JavaScript code:
```
class MyPromise extends Promise {
}

async function main() {
    const promise = new Promise()
    await promise

    const myPromise = new MyPromise()
    await myPromise
}
```

There is a potentially avoidable hint for the subclassed Promise. The original Promise has the flag `TypeFlags.Any` and the subclassed one has `TypeFlags.Object`.

<img src="https://private-user-images.githubusercontent.com/2276355/307611232-e20bad8d-e68b-465b-a59e-daa2b3756ef1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg4ODM0NjMsIm5iZiI6MTcwODg4MzE2MywicGF0aCI6Ii8yMjc2MzU1LzMwNzYxMTIzMi1lMjBiYWQ4ZC1lNjhiLTQ2NWItYTU5ZS1kYWEyYjM3NTZlZjEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMjVUMTc0NjAzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2U1MmE0Yjc5NjA5M2QyZTE1ZTk5ZjMxNDRhZWFkMzkxYWNjN2Q4ZTM4YTlkY2RhYjljNGU0ZjM1MTljZTQ5ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Gg1ute7FfAPCx8PeAxHIQmtfY0aN9m7fnAPR_AAeowY" width="50%">

These are possibly related:
https://github.com/microsoft/TypeScript/issues/45427
https://github.com/microsoft/TypeScript/issues/56978

### 📃 Motivating Example

In Visual Studio Code using the Quick Fix… feature to remove the `await` keyword can break existing code.

### 💻 Use Cases

1. To prevent incorrect hints.
2. Hints are incorrect.
3. Ignoring hints or disabling the "TypeScript and JavaScript Language Features" extension.

**Comment by Swimburger** [comment]
Is this still an issue, I don't see this quick fix hint in my VS Code instance.

<img width="343" alt="Image" src="https://github.com/user-attachments/assets/dad5c8d2-405e-4c42-aa8a-8faae5ef197d" />

**Comment by dnicolson** [comment]
Yes, it is still an issue with VS Code 1.98.2. It looks like the built-in extension is using TypeScript 5.9.0-dev.20250330.
